### ISSUE TITLE
willThrow and doThrow() accepting Class<? extends Exception> overloaded versions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#215]
```
Currently, org.mockito.BDDMockito.BDDMyOngoingStubbing#willThrow (and other 
variations of this method) accept instance of the exception to be thrown.

It would be nice to have a overloaded version that accept Class instance of 
exception. Mockito would create exception instance using reflection and default 
c-tor. The whole point of this improvement is to get rid of "Runtime exception 
not thrown" inspection/warning in some tools/IDEs (like IntelliJ IDEA).
```

Original issue reported on code.google.com by `nurkiew...@gmail.com` on 31 Aug 2010 at 9:05

**Comment by GoogleCodeExporter** [comment]
```
hmmmm interesting idea.
We'll consider this. The downside is that we need to extend the API.

In meantime you can create your own MyBDDMockito :)

Cheers!
```

Original comment by `szcze...@gmail.com` on 2 Sep 2010 at 5:18
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Could be done when/if we work on issue 146.
```

Original comment by `brice.du...@gmail.com` on 3 Dec 2010 at 2:49

**Comment by GoogleCodeExporter** [comment]
```
Issue 237 has been merged into this issue.
```

Original comment by `szcze...@gmail.com` on 12 Dec 2010 at 10:27

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 12 Dec 2010 at 10:27
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
Original comment by `brice.du...@gmail.com` on 23 Dec 2010 at 1:04
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
fixed by revision 31f383979d
```

Original comment by `brice.du...@gmail.com` on 25 Dec 2010 at 6:20
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 3 Jul 2011 at 12:43
- Added labels: **Milestone-Release1.9-rc1**
